modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/operations/util/OperationUtils.java

```diff
@@ -0,0 +1,506 @@
+     */
+    public List<Profile> generateDeviceOperationStatusObject(SyncmlDocument syncmlDocument) throws
+            NotificationManagementException, WindowsOperationException {
+
+        DeviceIdentifier deviceIdentifier = convertToDeviceIdentifierObject(
+                syncmlDocument.getHeader().getSource().getLocURI());
+        String lockUri = null;
+        Results result = syncmlDocument.getBody().getResults();
+
+        List<Profile> profiles = new ArrayList<>();
+        if (result != null) {
+            List<Item> results = result.getItem();
+            for (OperationCode.Info info : OperationCode.Info.values()) {
+                if (PluginConstants.OperationCodes.PIN_CODE.equals(info
+                        .name())) {
+                    lockUri = info.getCode();
+                }
+            }
+            for (Item item : results) {
+                for (OperationCode.Info info : OperationCode.Info.values()) {
+                    if (item.getSource().getLocURI().equals(info.getCode()) && info.name().equals(
+                            PluginConstants.OperationCodes.CAMERA_STATUS)) {
+                        Profile cameraProfile = new Profile();
+                        cameraProfile.setFeatureCode(PluginConstants.OperationCodes.CAMERA);
+                        cameraProfile.setData(item.getData());
+                        if (item.getData().equals(PluginConstants.SyncML.SYNCML_DATA_ONE)) {
+                            cameraProfile.setEnable(true);
+                        } else {
+                            cameraProfile.setEnable(false);
+                        }
+                        profiles.add(cameraProfile);
+                    }
+                    if (item.getSource().getLocURI().equals(info.getCode()) && info.name().equals(
+                            PluginConstants.OperationCodes.ENCRYPT_STORAGE_STATUS)) {
+                        Profile encryptStorage = new Profile();
+                        encryptStorage.setFeatureCode(PluginConstants.OperationCodes.ENCRYPT_STORAGE);
+                        encryptStorage.setData(item.getData());
+                        if (item.getData().equals(PluginConstants.SyncML.SYNCML_DATA_ONE)) {
+                            encryptStorage.setEnable(true);
+                        } else {
+                            encryptStorage.setEnable(false);
+                        }
+                        profiles.add(encryptStorage);
+                    }
+                    if (item.getSource().getLocURI().equals(info.getCode()) && info.name().equals(
+                            PluginConstants.OperationCodes.DEVICE_PASSWORD_STATUS)) {
+                        Profile encryptStorage = new Profile();
+                        encryptStorage.setFeatureCode(PluginConstants.OperationCodes.PASSCODE_POLICY);
+                        encryptStorage.setData(item.getData());
+                        if (item.getData().equals(PluginConstants.SyncML.SYNCML_DATA_ZERO)) {
```

Comment:
As a best practice, always compare the non-deterministic variable values against known constants.